bitcoin/src/blockdata/transaction.rs

```diff
@@ -1040,6 +1041,63 @@ impl Transaction {
     }
 }
 
+impl FromStr for Transaction {
+    type Err = ParseTransactionError;
+
+    fn from_str(hex: &str) -> Result<Self, Self::Err> {
+        let bytes = <Vec<u8> as hex::FromHex>::from_hex(hex)?;
+        Ok(consensus::encode::deserialize(&bytes)?)
+    }
+}
+
+impl fmt::Display for Transaction {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let v = encode::serialize(self);
+        fmt::LowerHex::fmt(&v.as_hex(), f)
+    }
+}
+
+/// Error parsing a [`Transaction`] from a hex string.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum ParseTransactionError {
+    /// Invalid hex string.
+    Hex(hex::HexToBytesError),
```

Comment:
Exposing this will make stabilization of primitives happen later. I'd prefer we don't do that.

Reply:
This is a pretty common one. I think we should take this opportunity to create a dedicated local opaque `HexError` type. I guess in a new module or something with the intention that we'd do the same for our other deps' errors (though I don't want to hold up this PR for it, since this is something we've really wanted for years).

Reply:
> Exposing this will make stabilization of primitives happen later.

Why? Because it makes the error part of the public API? We already expose the `hex` crate.

> I think we should take this opportunity to create a dedicated local opaque HexError type.

Do you mean that we should remove the `hex` crate from the public API all together?

eg: https://docs.rs/bitcoin/latest/bitcoin/key/enum.Error.html#method.from-3

Reply:
> though I don't want to hold up this PR for it

Fair.

> Do you mean that we should remove the hex crate from the public API all together?

Yes, at least for now.